### ISSUE TITLE
Validation of enums that are numbers stopped working in 1.8.0

### ISSUE THREAD

**Issue by fredriklengstrand** [redhat-developer/vscode-yaml#790]
## Describe the bug

It appears that validation of enums that are numbers stopped working in 1.8.0.
More concerning is that it also prevents other properties from validating when you use this property in the YAML file.

### Sample schema
```
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "type": "object",
    "properties": {
        "first": {
            "type": "string",
            "enum": ["a", "b"]
        },
        "second": {
            "type": "number",
            "enum": [1, 2]
        }
    }
}
```

## Sample YAML
```
# yaml-language-server: $schema=./test.schema.json
first: c
second: 3
```

## Expected Behavior

I should get an error that both properties are invalid.

## Current Behavior

No errors are shown for either of the properties.

## Steps to Reproduce

Use the provided sample files and try commenting out / enabling the `second` property and you will see that the validation for `first` starts working / stops working.

## Environment
- [ ] Windows
- [X] Mac
- [ ] Linux
- [ ] other (please specify)

**Comment by beanaroo** [comment]
I was curious as I'm trying this extension out for the first time and it seems that enum of type `number` is causing the issue.

**Comment by msivasubramaniaan** [comment]
> I was curious as I'm trying this extension out for the first time and it seems that enum of type `number` is causing the issue.

Yes the issue related to `number ` type. And the issue was addressed and the fix will be available on the new v1.9.0
![image](https://user-images.githubusercontent.com/93245779/177283164-42af9901-92c4-48fa-ab3d-034bd7394b39.png)

**Comment by msivasubramaniaan** [comment]
@beanaroo and @fredriklengstrand as new YAML v1.9.1 was released and please check the mentioned issue on the corresponding version.

**Comment by fredriklengstrand** [comment]
@msivasubramaniaan Thank you for looking into this. I have done some testing and there still seems to be some issues left when  you have defined a number enum and then for example input a string in  there instead. That stops all validation from working. With the schema I sent over before, this stops all validation:

```
# yaml-language-server: $schema=./test.schema.json
first: c
second: a
```
